civis_jupyter_notebooks/log_utils.py

```diff
@@ -15,13 +15,17 @@ def setup_stream_logging():
     logger.propagate = False
     # make sure to grab orig stderr since things seem to get redirected a bit
     # by the notebook and/or ipython...
-    ch = logging.StreamHandler(stream=sys.__stderr__)
+    error_handler = logging.StreamHandler(stream=sys.__stderr__)
+    info_handler = logging.StreamHandler(stream=sys.__stdout__)
     formatter = logging.Formatter(
         fmt='[%(levelname).1s %(asctime)s %(name)s] %(message)s',
         datefmt="%H:%M:%S")
-    ch.setLevel(logging.INFO)
-    ch.setFormatter(formatter)
-    logger.addHandler(ch)
+    error_handler.setLevel(logging.WARNING)
```

Comment:
I'm a bit confused - does this mean that warn-level logs from the notebook are handled by the `error_handler` and then sent to stderr?  Do we need another handler for error-level logs, or other levels?

Reply:
So from my research, log levels and stream handlers are separated concepts in python. When you call `setLevel` on a logger (which I think is actually a stream handler, or a collection of stream handlers?), you are telling that logger/streamHandler to pay attention to all logs at the specified level or higher. So the error_handler we create here will send all logs of severity `WARNING`(30) or higher to `sys.__stderr__` and the info_handler will send all logs of severity `INFO`(20) or higher to `sys.__stdout__`. I think the way that this is configured right now, warning+ logs will be logged both to stderr and stdout.